Remove pulse defaults for old fake 20q backends

mtreinish opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
-->

### Summary

The pulse defaults payload for the almaden, johannesburg, and singapore
fake backends are collectively taking up > 50% of the total disk
footprint of an installed copy of Qiskit. This is because the pusle
defaults for these backends predate the use the of parameterized pulse
definitions and contain arrays of sampled waveforms for the pulse
definitions. This exceedingly large file size slows down the entire
package and and makes it slower to install. When weighing the potential
value these defaults files provide for simulating and compiling with
pulse awareness for long retired devices against the real cost of
having the files included in the package there isn't a reason to keep
the files in the tree any longer. This commit removes the pulse
defaults files for these fake backends to greatly shrink the installed
package size of qiskit-terra.

### Details and comments

jakelishman reviewed (dismissed):
Absolutely fine by me to get rid of these.  They're not so heavy for the download of the wheels (due to the compression), but it just seems like good bookkeeping to clear out cruft from devices that have been offline for however long.

jakelishman commented:
Some of the tests explicitly use the Almaden, Johannesburg and Singapore fakes in part because they had pulse defaults - those will need changing over.

nkanazawa1989 reviewed (dismissed):
I'm also fine with removing them. These pulse data could be used for pulse simulation, but Hamiltonian parameters in configuration are usually off from calibration and useless apart from testing of the visualization module (because we don't update Hamiltonian with calibrations).

mtreinish commented:
ok, I've updated the tests to no longer rely on these backends having pulse defaults present. Hopefully things will pass now.

mtreinish commented:
The tutorials failure should be fixed once: https://github.com/Qiskit/qiskit-tutorials/pull/1356 merges

jakelishman commented on releasenotes/notes/remove-pulse-defs-old-20q-4ed46085b4a15678.yaml:
```diff
@@ -0,0 +1,21 @@
+  - |
+    The fake backend objects :class:`~.FakeJohannesburg`,
+    :class:`~.FakeJohannesburgV2`, :class:`~.FakeAlmaden`,
+    :class:`~.FakeAlmadenV2, :class:`~.FakeSingapore`, and
```
```suggestion
    :class:`~.FakeAlmadenV2`, :class:`~.FakeSingapore`, and
```

jakelishman reviewed (approved):
Looks good to me - the diff looks cleaner without the +1,800 file as well!  I'm never very sure if we _should_ be making assertions about the exact forms of those giant reprs anyway.

mtreinish commented:
> Looks good to me - the diff looks cleaner without the +1,800 file as well! I'm never very sure if we _should_ be making assertions about the exact forms of those giant reprs anyway.

Heh, that took me longer to track down than I care to admit. My default editor config was changing the whitespace and I thought I somehow broke the output as part of the unrelated change which is why I changed the backend and added that file. But once I figured out what I did wrong it was easy to fix.
